interface/src/app/signature/enriched_signatures.js

```diff
@@ -0,0 +1,195 @@
+      }
+    },
+    data: {pageTitle: 'Enriched Signatures'}
+  });
+}])
+
+.controller('EnrichedSignaturesCtrl', ['$stateParams', 'Signature', '$log',
+  'errGen', 'Participation', 'MathFuncts', 'Gene',
+  function EnrichedSignatureController($stateParams, Signature, $log, errGen,
+                                       Participation, MathFuncts, Gene) {
+    var self = this;
+    self.isValidModel = false;
+    // Do nothing if the model ID in URL is falsey. The error will be taken
+    // care of by "<ml-model-validator>" component.
+    if (!$stateParams.mlmodel) {
+      return;
+    }
+
+    self.statusMessage = 'Connecting to the server ...';
+
+    // Do nothing if no genes are specified in URL.
+    if (!$stateParams.genes || !$stateParams.genes.split(',').length) {
+      self.statusMessage = 'No genes are specified.';
+      self.enrichedSignatures = [];
+      return;
+    }
+
+    var userSelectedGenes = $stateParams.genes.split(',')
+
+    Signature.get(
+      {mlmodel: self.modelInUrl, limit: 0},
+      function success(response) {
+        self.signatures = response.objects;
+        self.statusMessage = '';
+      },
+      function error(err) {
+        var message = errGen('Failed to get signatures: ', err);
+        $log.error(message);
+        self.statusMessage = message + '. Please try again later.';
+      }
+    );
+
+    // For the moment, the total number of genes in our universe
+    // to perform this enrichment analysis, is the total number of
+    // Entrez IDs in our database for Pseudomonas aeruginosa.
+    Gene.get(
+      {},
+      function success(response) {
+        self.geneNum = response.meta.total_count;
+      },
+      function error(err) {
+        var message = errGen('Failed to get total gene number: ', err);
+        $log.error(message);
+        self.statusMessage = message + '. Please try again later.';
+      }
+    );
+
+    self.pValueCutoff = 0.05;
+    var pValueSigDigits = 3;
+
+    // This is an object, of signatures per participation type
+    var signtrsPerPT = {};
+
+    self.enrichedSignatures = {};
+
+    // This is the main function that calculates the signature enrichments.
+    // It calculates the enrichment for each signature that has participatory
+    // genes which were in the url genes (the genes searched).
+    var calculateEnrichments = function(totalGeneNum, signature, cutoff) {
+      // For the moment, this is the total number of Entrez IDs for
+      // Pseudomonas in the database.
+      var N = totalGeneNum;
+
+      // This will be the number of genes in the url (which are the genes the
+      // user selected in the previous page)
+      var m = userSelectedGenes.length;
+
+      var pValueArray = [];
+
+      var signatureIds = Object.keys(signatures);
+
+      for (var i = 0; i < signatureIds.length; i++) {
+        var signatureId = signatureIds[i];
+        var genes = signatures[signatureId];
+        var n = genes.length;
```

Comment:
@rzelayafavila: Is `n` the total number of heavy genes associated with `signatureId`?

Reply:
Hi @dongbohu - Yes, it is meant to be the total number of genes that have participation in a signature (for a specific type of participation, can be high weight genes or another type of participation).